src/database_connection.py

```diff
@@ -36,7 +36,7 @@ def build_db_if_needed(self):
 
     def uploaded_files(self):
         result = self.safe_query(self.q.get_uploaded_files)
-        return [(row['s3filename'], row['destination']) for row in result]
+        return {"%s - %s" % (row['s3filename'], row['destination']) for row in result}
```

Comment:
There's no need to do the "%s - %s" string mapping, a set can have keys that are a tuple of strings.

Reply:
Hum, I tested it locally, looks like with 2.3 millions uploaded file and 1k to-process csv file, it is only 10% faster. I'll apply the tuple idea, and leave the API as it was. So we don't need to change the test case then.